test/ShallowWrapper-spec.jsx

```diff
@@ -1006,6 +1006,189 @@ describe('shallow', () => {
     });
   });
 
+  describe('.invoke(eventName, ..args)', () => {
+    it('should return the handlers return value', () => {
+      const spy = sinon.stub().returns(123);
+      class Foo extends React.Component {
+        render() {
+          return (
+            <a onClick={spy}>foo</a>
+          );
+        }
+      }
+
+      const wrapper = shallow(<Foo />);
+      const value = wrapper.invoke('click');
+
+      expect(value).to.equal(123);
+      expect(spy.called).to.equal(true);
+    });
+
+    it('should invoke event handlers without propagation', () => {
+
+      class Foo extends React.Component {
+        constructor(props) {
+          super(props);
+          this.state = { count: 0 };
+          this.incrementCount = this.incrementCount.bind(this);
+        }
+
+        incrementCount() {
+          this.setState({ count: this.state.count + 1 });
+        }
+
+        render() {
+          return (
+            <div onClick={this.incrementCount}>
+              <a
+                className={`clicks-${this.state.count}`}
+                onClick={this.incrementCount}
+              >foo</a>
```

Comment:
nbd, especially in a test, but i'd expect the linter to error out here and require:
```jsx
<a
  className={`clicks-${this.state.count}`}
  onClick={this.incrementCount}
>
  foo
</a>
```